### PR TITLE
[WFLY-6409]:  Update default domain configuration to use http-remoting

### PR DESCRIPTION
**Pull request by ehsavoie**
Updating default domain configurations.

Jira: https://issues.jboss.org/browse/WFLY-6409
JBEAP: https://issues.jboss.org/browse/JBEAP-9093
Supersede: https://github.com/wildfly/wildfly/pull/8798

### CODE REVIEW

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java [hunk 1]
```diff
@@ -91,55 +92,55 @@ private static File hackFixHostConfig(File hostConfigFile, String hostName, Stri
...
                 boolean processedOpt = false;
                 while (line != null) {
-                    int start = line.indexOf("<host");
-                    if (start >= 0 && !line.contains(" name=")) {
-                        StringBuilder sb = new StringBuilder();
-                        sb.append("<host name=\"");
-                        sb.append(hostName);
-                        sb.append('"');
-                        sb.append(line.substring(start + 5));
+                int start = line.indexOf("<host");
```
Comment (jmesnil) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java]: code is not properly indented

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/common/EJBManagementUtil.java [hunk 2]
```diff
@@ -267,7 +267,7 @@ public static void createRemoteOutboundConnection(final ModelControllerClient mo
             addRemoteOutboundConnection.get("outbound-socket-binding-ref").set(outboundSocketRef);
             addRemoteOutboundConnection.get(SECURITY_REALM).set("PasswordRealm");
             addRemoteOutboundConnection.get("username").set("user1");
-            addRemoteOutboundConnection.get("protocol").set("http-remoting");
+            addRemoteOutboundConnection.get("protocol").set("remote+http");
```
Comment (jmesnil) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/common/EJBManagementUtil.java]: Is that related to domain configuration?

**Review by jmesnil — COMMENTED** [review]
What's the scope of WFLY-6409? 

Reading the issue description, my understanding is that `remote-http` should be used for domain mode, i.e. domain hosts communicate between themselves using `remote-http` instead of the native `remote`.

However this PR goes beyond that and replaces most (or all?) use of `remote` with `remote-http`, including remote EJB communications, etc.

Could you clarify the scope of the issue?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/common/EJBManagementUtil.java (hunk 2 above)
Comment (ehsavoie) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/common/EJBManagementUtil.java]: http-remoting is deprecated in favor of remote+http

**Comment by rsvoboda** [comment]
@simkam @mjurc please take a look into this PR

**Review by simkam — APPROVED** [review]

**Comment by kabir** [comment]
Retest this please

File: feature-pack/src/main/resources/configuration/host/host.xml [hunk 3]
```diff
@@ -63,6 +60,8 @@
 
     <domain-controller>
         <local/>
+        <!-- Alternative remote domain controller configuration with a host and port -->
+        <!-- <remote protocol="remote+http" host="${jboss.domain.master.address}" port="${jboss.domain.master.port:9990}" security-realm="ManagementRealm"/> -->
```
Comment (bstansberry) [inline feature-pack/src/main/resources/configuration/host/host.xml]: No comments in our standard config files.

File: servlet-feature-pack/src/main/resources/configuration/host/host.xml [hunk 4]
```diff
@@ -63,6 +60,8 @@
 
     <domain-controller>
         <local/>
+        <!-- Alternative remote domain controller configuration with a host and port -->
+        <!-- <remote protocol="remote+http" host="${jboss.domain.master.address}" port="${jboss.domain.master.port:9990}" security-realm="ManagementRealm"/> -->
```
Comment (bstansberry) [inline servlet-feature-pack/src/main/resources/configuration/host/host.xml]: No comments in standard config files.

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/ejb/ssl/SSLEJBRemoteClientTestCase.java [hunk 5]
```diff
@@ -126,13 +126,13 @@ public void prepareServerOnce() throws Exception {
             log.trace("*** starting server");
             container.start(DEFAULT_JBOSSAS);
             final ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient();
-            managementClient = new ManagementClient(client, TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getServerPort(), "http-remoting");
-            log.trace("*** will configure server now");
+            managementClient = new ManagementClient(client, TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getServerPort(), "remote+http");
+            log.info("*** will configure server now");
```
Comment (bstansberry) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/ejb/ssl/SSLEJBRemoteClientTestCase.java]: log.trace

File: testsuite/shared/src/main/resources/org/jboss/as/test/integration/web/sso/resources/web-form-auth.xml [hunk 6]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE web-app PUBLIC "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN" "http://java.sun.com/dtd/web-app_2_3.dtd">
 <web-app>
-    <distributable/>
```
Comment (bstansberry) [inline testsuite/shared/src/main/resources/org/jboss/as/test/integration/web/sso/resources/web-form-auth.xml]: Unrelated

**Review by bstansberry — CHANGES_REQUESTED** [review]
There are minor changes required.

Beyond those, the scope of this issue has been expanded beyond the original change of the intra-domain interface. All the http-remoting -> remote+http changes outside of the ones related to intra-domain communication would have been better in a separate JIRA / PR. But since the only non-test code affected by that is jdr/CommandLineMain and if there had been a separate PR for those change I'd have approved it, I can live with it here.

This needs ok from @jmesnil.

**Comment by kabir** [comment]
I am deferring this to WF15 - there is too much other stuff going on now for 14 :)

**Comment by kabir** [comment]
@jamezp @jmesnil Removing the 15.x label. It should be ok if you guys are ok with it

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/web/ssl/HTTPSWebConnectorTestCase.java [hunk 7]
```diff
@@ -320,8 +320,8 @@ public void stopContainer() throws Exception {
         deployer.undeploy(APP_CONTEXT);
         final ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient();
         final ManagementClient managementClient = new ManagementClient(client, TestSuiteEnvironment.getServerAddress(),
-                TestSuiteEnvironment.getServerPort(), "http-remoting");
-        LOGGER.trace("*** reseting test configuration");
+                TestSuiteEnvironment.getServerPort(), "remote+http");
+        LOGGER.info("*** reseting test configuration");
```
Comment (jamezp) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/web/ssl/HTTPSWebConnectorTestCase.java]: Was there a reason to change this to info instead of trace?

**Review by jamezp — APPROVED** [review]
This seems fine to me.

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/web/ssl/HTTPSWebConnectorTestCase.java (hunk 7 above)
Comment (ehsavoie) [reply testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/web/ssl/HTTPSWebConnectorTestCase.java]: I don't remember except that the test was failing at some point

Comment (kabir) [reply testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/web/ssl/HTTPSWebConnectorTestCase.java]: @ehsavoie I think make it trace again

**Comment by kabir** [comment]
@ehsavoie The elytron job has failed a few times with similar errors, so I presume it is caused by this PR

https://ci.wildfly.org/viewType.html?buildTypeId=WF_PullRequest_LinuxElytro&tab=buildTypeHistoryList&branch_WF_PullRequest=pull%2F11351

**Comment by ehsavoie** [comment]
@kabir the problem lies in the fact that there is no native management interface so the cli script fails in domain mode
